1. Brokers/1. Brokers.md

```diff
@@ -254,8 +253,7 @@ namespace OtripleS.Web.Api.Brokers.Storages
             return @object;
         }
 
-        private ValueTask<IQueryable<T>> SelectAllAsync<T>() where T : class =>
-            await ValueTask.FromResult(this.Set<T>());
+        private ValueTask<IQueryable<T>> SelectAllAsync<T>() where T : class => this.Set<T>();
```

Comment:
Please correct me if I am wrong. 

this.set<T> returns an IQueryable<T> where a ValueTask<IQueryable<T>> is expected. 

we can change it to below? 

private ValueTask<IQueryable<T>> SelectAllAsync<T>() where T : class => 
    ValueTask.FromResult(this.Set<T>());

We can remove the await, as its used to asynchronously wait for the valuetask to complete, however because ValueTask.FromResult immediately completes the task, await keyword is unnecessary.

Reply:
> Please correct me if I am wrong.
> 
> this.set returns an **IQueryable where a ValueTask<IQueryable>** is expected.
> 
> we can change it to below?
> 
> private ValueTask<IQueryable> SelectAllAsync() where T : class => ValueTask.FromResult(this.Set());
> 
> We can remove the await, as its used to asynchronously wait for the valuetask to complete, however because ValueTask.FromResult immediately completes the task, await keyword is unnecessary.

@hassanhabib ^^^^